### DISCUSSION TITLE
Manually readd device

### DISCUSSION THREAD

**Discussion by timcanty** [Ylianst/MeshCentral#6439] [General]
Hi, 
Think this is a long shot, To cut a long storey short a while ago, we had to rebuild our database, and reinstall all the agents across devices because all the device groups etc didn't exist. Which hasn't been an issue.

However we have a device out in the far sticks of the country we haven't managed to get reinstalled yet, we can see it attempting in the logs on our server from the old agent to connect, i assume the issue is the group doesn't exist.

Is there anyway we can manually add this into the database or something, to allow it to connect so we can get on it. 
We have a long string in the logs, which i assume would map to some kind of id in the database.

Agent connected with invalid domain/mesh, holding connection (*.*.*.*:49717, mesh//***********************************************************).

just it would be fantastic to be able to get on this device to fix an issue with it, but at the moment we don't have the means without spending hours on the road to go uninstall/reinstall the agent.

any ideas much appreciated, but if its a no go that's fine.

Kind Regards

**Comment by timcanty** [comment]
I've just discovered this, [https://github.com/Ylianst/MeshCentral/issues/2095](https://github.com/Ylianst/MeshCentral/issues/2095) might aid is, will dig some more

**Reply by si458** [reply]
this is exactally what i was about to post but was just checking the source code first!
https://github.com/Ylianst/MeshCentral/issues/2095#issuecomment-749406399
`"orphanAgentUser": "myusernamehere"`

**Reply by timcanty** [reply]
Thanks simon, do you happen to know if its an odic user from azure, what we put in the myusername, is it the user identifier? before i trial an error it

**Reply by si458** [reply]
if you login as admin, then goto into the users tab and select ur user,
you will see the `user identifier`,
so your username would be something like `user//myuser`
you would just enter the `myuser` bit
(you remove the `user//` bit OR if you have mutliple domains `user/domain2/` if that makes sense)

**Reply by timcanty** [reply]
Brilliant, that worked a treat,
so i copied this part (~oidc:***********************************) out of the user identifier and stuck that in the config and that got the device online. 
Thank you so much.

**Reply by si458** [reply]
glad it worked!
